### ISSUE TITLE
macOS Runners

### ISSUE THREAD

**Issue by ChrisEdson** [philips-labs/terraform-aws-github-runner#2069]
Hi there -

Is it possible to support macOS runners using this module?

Technically, GitHub [states that it is possible](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#macos) - and AWS also has mac instances that it can spin up.

Thanks!

**Comment by npalm** [comment]
Yes this should certainly be possible. We have crafted the module for Amazon Linux back in the days. Later support was added for multiple architectures, linux distros, and windows. All with the great support of the community. 

Differences betwoon OS are in general
- The base AMI (AWS)
- The launch script to setup the instance and install the runner. Aka the user_data script

We certainly welcome PR's to support MacOS as well.

**Comment by s0undt3ch** [comment]
FYI, besides the AWS Mac VMs costing a lot more, their minimum reservation is 24 hours.
So, ephemeral Mac runners is off the table.

**Comment by 4sudiptodas** [comment]
We are trying to re-use the module for macos runners, by changing the ami filter and user data. However there is a variable runner_os which can only have (linux/windows) as valid inputs. How do we workaround this? any suggestions @npalm

**Comment by npalm** [comment]
Would be great when you are able to spport macos. I think it requires some refactoring the terraform module to allow multiple values. As well the model the scal-up is creating and scale-down is destroying is not fitting the model for OSX runners. How are you approaching this?

**Comment by davidnewhall** [comment]
> However there is a variable `runner_os` which can only have (linux/windows) as valid inputs.

Set `runner_os` to `linux`, set `runner_disable_default_labels` to `true` and set these variables appropriately:
- `userdata_content` 
- `ami`
- `runner_extra_labels` (set all the labels you need including `self-hosted`)
- `instance_types` (`["mac1.metal"]`, etc)
- `minimum_running_time_in_minutes` (should be 1440)

The only other place `runner_os` seems to be used is in the cloudwatch logs configuration, so just use the same paths as linux for logs and it should be fine.

The hardest part about this is what to put into `userdata_content`. You can start with the linux template, but then you quickly realize is depends on things like `svc.sh` that you have no familiarity with. Then you say to yourself "sheesh, I have no idea how to actually test this script effectively." It gets daunting real quick, don't ask how I know. 

Once you figure out what works in userdata_content you should then back-port your work into the custom AMI so the runners spin up quickly.  And at the same time, open a PR against this repo to share your success with everyone else. <3

EDIT: now I see the open PR haha. Wonderful. I'll test it shortly.

**Comment by edersonbrilhante** [comment]
@davidnewhall Closing as already implemented
